embedded/sql/stmt.go

```diff
@@ -1779,15 +1778,12 @@ func (c *Cast) getConverter(src, dst SQLValueType) (converterFunc, error) {
 }
 
 func (c *Cast) inferType(cols map[string]ColDescriptor, params map[string]SQLValueType, implicitDB, implicitTable string) (SQLValueType, error) {
-	valType, err := c.val.inferType(cols, params, implicitDB, implicitTable)
+	_, err := c.val.inferType(cols, params, implicitDB, implicitTable)
 	if err != nil {
 		return AnyType, err
 	}
 
-	_, err = c.getConverter(valType, c.t)
-	if err != nil {
-		return AnyType, err
-	}
+	// val type may be restrtiicted by compatible conversions, but multiple types may be compatible...
```

Comment:
```suggestion
	// val type may be restrticted by compatible conversions, but multiple types may be compatible...
```

Reply:
the typo was so bad that it required two fixes... restrtiicted -> restrticted -> restricted